unstructured/ingest/v2/processes/connectors/salesforce.py

```diff
@@ -0,0 +1,293 @@
+                raise ValueError(f"{record_type} not currently an accepted Salesforce category")
+
+            try:
+                # Get ids from Salesforce
+                records = client.query_all_iter(
+                    f"select Id, SystemModstamp, CreatedDate, LastModifiedDate from {record_type}",
+                )
+                for record in records:
+                    files_list.append(
+                        FileData(
+                            connector_type=CONNECTOR_TYPE,
+                            identifier=record["Id"],
+                            source_identifiers=SourceIdentifiers(
+                                filename=record["Id"],
+                                fullpath=f"{record['attributes']['type']}/{record['Id']}",
+                            ),
+                            metadata=DataSourceMetadata(
+                                url=record["attributes"]["url"],
+                                version=record["SystemModstamp"],
+                                date_created=record["CreatedDate"],
+                                date_modified=record["LastModifiedDate"],
+                                record_locator={"id": record["Id"]},
+                            ),
+                            additional_metadata={"record_type": record["attributes"]["type"]},
+                        )
+                    )
+            except SalesforceMalformedRequest as e:
+                raise SalesforceMalformedRequest(f"Problem with Salesforce query: {e}")
+
+        return files_list
+
+    def run(self, **kwargs: Any) -> Generator[FileData, None, None]:
+        for f in self.list_files():
+            yield f
+
+
+@dataclass
+class SalesforceDownloaderConfig(DownloaderConfig):
+    pass
+
+
+@dataclass
+class SalesforceDownloader(Downloader):
+    connection_config: SalesforceConnectionConfig
+    download_config: SalesforceDownloaderConfig = field(
+        default_factory=lambda: SalesforceDownloaderConfig()
+    )
+    connector_type: str = CONNECTOR_TYPE
+
+    def get_file_extension(self, record_type) -> str:
+        if record_type == "EmailMessage":
+            extension = ".eml"
+        elif record_type in ["Account", "Lead", "Case", "Campaign"]:
+            extension = ".xml"
+        else:
+            raise MissingCategoryError(
+                f"There are no categories with the name: {record_type}",
+            )
+        return extension
+
+    def get_download_path(self, file_data: FileData) -> Path:
+        record_file = file_data.identifier + self.get_file_extension(
+            file_data.additional_metadata["record_type"]
+        )
+        return Path(self.download_dir) / file_data.additional_metadata["record_type"] / record_file
+
+    def _xml_for_record(self, record: OrderedDict) -> str:
+        """Creates partitionable xml file from a record"""
+        import xml.etree.ElementTree as ET
+
+        def flatten_dict(data, parent, prefix=""):
```

Comment:
This method already exists, could that be reused? [unstructured/staging/base.py](https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/staging/base.py#L224)

Reply:
@rbiseck3 I can't even tell what this method does to the file. I tried running it and printing out the before and after and it looked like no change. But If I leave it out, the resulting file is way different.

I assume it is modifying the OrderedDict inplace...